verification.kt

```diff
@@ -0,0 +1,38 @@
+import java.math.BigInteger
+import java.nio.charset.StandardCharsets
+import java.time.Instant
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+
+fun generateUserIdHash(secretKey: String, userId: String): String {
+    val now = Instant.now().epochSecond.toString()
+    val secret = secretKey.toByteArray(StandardCharsets.UTF_8)
+    val secretKeySpec = SecretKeySpec(secret, "HmacSHA256")
+
+    val userAndTime = "$userId-$now"
+    val userAndTimeBytes = userAndTime.toByteArray(StandardCharsets.UTF_8)
+
+    val mac = Mac.getInstance("HmacSHA256")
+    mac.init(secretKeySpec)
+    val hash = BigInteger(1, mac.doFinal(userAndTimeBytes))
+
+    return String.format("%032x-%s", hash, now)
+}
+
+fun main() {
+    // Retrieve the secret key from environment variables
+    val secretKey = System.getenv("CHAMELEON_VERIFICATION_SECRET")
+        ?: throw IllegalArgumentException("Environment variable CHAMELEON_VERIFICATION_SECRET is not set")
+
+    val userId = "your_user_id"
+
+    // Generate the UID hash
+    val uidHash = generateUserIdHash(secretKey, userId)
+    println("Generated UID Hash: $uidHash")
+}
```

Comment:
I think we can leave this part and move the `System.getenv("CHAMELEON_VERIFICATION_SECRET")` into the function (then without the secret key argument)